src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -189,8 +207,47 @@ class BaseOptionsSelector extends Component {
         return defaultIndex;
     }
 
+    /**
+     * When pagination is enabled,
+     * maps sections to render only allowed count of them.
+     *
+     * @returns {Objects[]}
+     */
+    sliceSections() {
+        if (!_.isNumber(CONST.OPTIONS_SELECTOR_PAGE_LIMITATION)) {
+            return this.props.sections;
+        }
+
+        return _.map(this.props.sections, (section) => {
+            if (_.isEmpty(section.data)) {
+                return section;
+            }
+
+            return {
+                ...section,
+                data: section.data.slice(0, CONST.OPTIONS_SELECTOR_PAGE_LIMITATION * lodashGet(this.state, 'paginationPage', 1)),
+            };
+        });
+    }
+
+    /**
+     * Calculates all exactly visible options of sections.
```

Comment:
If I understand correctly, this is calculating how many items are currently visible is that right? For example, if there are 3000 items and we are currently showing 1000 of them (because the person has pressed "show more" once), this would return 1000?

Reply:
Yes, correct. This number is based on all items that we pass to **OptionsList** to render.

Reply:
Ok got it! This is my suggestion then. 
```
    /**
     * Calculates all currently visible options based on the sections that are currently being shown 
     * and the number of items of those sections.  
     *
     * @returns {Number}
     */
    getAllVisibleOptionsCount() {
```

Reply:
Done - c4f1f63.